core/arch/riscv/include/kernel/thread.h

```diff
@@ -0,0 +1,418 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2014, STMicroelectronics International N.V.
+ * Copyright (c) 2016-2017, Linaro Limited
+ * Copyright (c) 2020-2021, Arm Limited
+ * Copyright 2022 NXP
+ */
+
+#ifndef KERNEL_THREAD_H
```

Comment:
If I understand the arch model we're putting in place, I think this file should be called `core/arch/riscv/include/kernel/thread_arch.h` and it should include `<kernel/thread.h>`. Then you can get rid of many definitions below.